compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmValueGen.java

```diff
@@ -472,55 +429,14 @@ private void createRecordConstructor(ClassWriter cw, String argumentClass, Strin
         // invoke `super(type)`;
         mv.visitMethodInsn(INVOKESPECIAL, MAP_VALUE_IMPL, JVM_INIT_METHOD, argumentClass, false);
 
-        mv.visitInsn(RETURN);
-        JvmCodeGenUtil.visitMaxStackForMethod(mv, JVM_INIT_METHOD, className);
-        mv.visitEnd();
-    }
-
-    // TODO: remove this method, logic moved to createInstantiateMethod, see #23012
-    private void createRecordInitWrapper(ClassWriter cw, String className, BIRNode.BIRTypeDefinition typeDef) {
-
-        MethodVisitor mv = cw.visitMethod(ACC_PUBLIC + ACC_STATIC , RECORD_INIT_WRAPPER_NAME, RECORD_INIT_WRAPPER,
-                null, null);
-        mv.visitCode();
-        // load strand
-        mv.visitVarInsn(ALOAD, 0);
-        // load value
-        mv.visitVarInsn(ALOAD, 1);
-
-        // Invoke the init-functions of referenced types. This is done to initialize the
-        // defualt values of the fields coming from the referenced types.
-        for (BType bType : typeDef.referencedTypes) {
-            BType typeRef = JvmCodeGenUtil.getReferredType(bType);
-            if (typeRef.tag != TypeTags.RECORD) {
-                continue;
+        if (!JvmCodeGenUtil.isAnonType(typedef)) {
+            Label label = new Label();
+            if (typedef.pos != null && typedef.pos.lineRange().endLine().line() != 0x80000000) {
```

Comment:
Shall we make `0x80000000` a constant and put in JVMConstants and use the same in the JvmCodeGenUtil